woocommerce/woocommerce-android
# Join order addons data with global addons

Summary
==========
Fixes issue #4567 by making it possible to join Global Add-ons data within the Ordered Add-ons view. Since the Global Add-ons data isn't something available inside Order and Product models, we need to do a fetch operation that may lead to loading times, so this PR also introduces skeleton loading mechanics to the view.

Screenshots
==========
| Ordered Add-ons view loading skeleton  | Ordered Add-ons View containing Global Add-ons |
| ------------- | ------------- |
| ![Screenshot_20210829_182747](https://user-images.githubusercontent.com/5920403/131265930-383c9754-4472-46b3-a681-99a9b0cfa04a.png) | ![Screenshot_20210829_182733](https://user-images.githubusercontent.com/5920403/131265929-7c415714-a566-49d0-8be6-7754dc7fd00b.png) |

How to Test
==========
Prerequisites: Have your site with the add ons plugin installed, configured, and with at least one product with add-ons and at least one order of the configured product including Global Add-ons. 

⚠️ Note that you should configure both Product-specific and Global add-ons in your store in order to test this improvement properly.

1 - Launch the Woo app.
2 - Navigate to an order with a Product configured with Add-ons, verify if the `View Add-ons` Button is showing up for products that contain Add-ons inside the current Order, but doesn't show up for Products without Add-ons for that order or Products that don't have Add-ons configured at all.
3 - Click on `View Add-ons` and verify if the Ordered Addons view is displayed with all the expected data, including Global Add-ons data.


Update release notes:
